src/main/scala/Silicon.scala

```diff
@@ -296,6 +293,28 @@ class Silicon(val reporter: Reporter, private var debugInfo: Seq[(String, Any)]
     failures
   }
 
+  private def failureFilterAndGroupingCriterion(f: Failure): String = {
+    // apply transformers if available:
+    val transformedError = f.message match {
+      case e: AbstractVerificationError => e.transformedError()
+      case e => e
+    }
+    // create a string that identifies the given failure:
+    transformedError.readableMessage(withId = true, withPosition = true)
+  }
+
+  private def failureSortingCriterion(f: Failure): Tuple2[Int, Int] = {
```

Comment:
I believe that IntelliJ recommends `(Int, Int)` over `Tuple2[Int, Int]`. I at least prefer the former; less syntactic clutter.